### ISSUE TITLE
Handle nested array fields with objects

### ISSUE THREAD

**Issue by Fryuni** [facebook/relay#3599]
We have 2 dimension arrays our GraphQL API like so:
```GraphQL
type TableCell {
  value: Int!
  label: String!
}

type Table {
  data: [[TableCell!]!]!
}
```

When we receive the data, RelayNormalizer is turning it into:
```js
{
  data: [
    {value: undefined, label: undefined},
    {value: undefined, label: undefined},
    {value: undefined, label: undefined},
    {value: undefined, label: undefined},
  ]
}
```

The normalizer is trying to read the `value` and `label` from the seconds layer of Arrays, which don't exist, so they become `undefined`.

**Comment by Fryuni** [comment]
For those that reach the same problem, if you control the schema, just split it into an extra type. For us we did this:
```GraphQL
type TableRow {
  cells: [TableCell!]!
}

type Table {
  rows: [TableRow!]!
}
```

We haven't found a workaround for when we can't control the schema

**Comment by ValdemarGr** [comment]
I don't think that this issue should be closed. The author found a workaround, not a fix.

**Comment by pluong** [comment]
What if we don't have the ability to update the schema which is controlled by another party? This issue seems like something should be fixed as it's a basic ability to deserialize data.

**Comment by Fryuni** [comment]
> This issue seems like something should be fixed as it's a basic ability to deserialize data.

Overall relay is **very** opinionated, most times we agree with the imposed opinions, others we strongly disagree. We don't expect things like this to be fixed since it is not "the correct way" by some unknown being's definition.

We could open PRs for it, but a lot of its internals are very much designed _for meta_, it includes a bunch of specific behavior for them (literally comments in the codebase that are supposed to document a function just link to internal facebook docs that are employee-only), so it is basically a minefield. We use a little piece of a big machine that happens to be useful for us and who knows how the big machine depends on it or vice versa.

We could also migrate to a different React GraphQL client. Even tho nowadays there are good alternatives, their design is so different from relay that it would basically require rewriting your whole frontend. It is easier to make wrappers around relay to hide bad opinions such as this than to move to other clients, so we just deal with it.

> What if we don't have the ability to update the schema which is controlled by another party?

Make a wrapper server around it and expose a different schema. That is what we did.
